### ISSUE TITLE
Warn when checking whether unsigned is greater that negative number.

### ISSUE THREAD

**Issue by m93a** [dotnet/roslyn#23473]
**Version Used**: N/A

**Steps to Reproduce**:

```c#
for(uint i = length; i >= 0; i--) something();
for(uint i = length; i > -1; i--) something();
```

**Expected Behavior**: Show a warning that unsigned is always greater than a negative number.

**Actual Behavior**: No warning appears.

**Comment by CyrusNajmabadi** [comment]
This sounds like something an analyzer could provider.
